backend/src/main/java/com/stock/admin/model/entity/Product.java

```diff
@@ -0,0 +1,227 @@
+     * @param shopCode  the shop code
+     * @param createdAt the created at
+     * @param updatedAt the upated at
+     */
+    public Product(String code, String name, String shopCode, Date createdAt, Date updatedAt) {
+        this.code = code;
+        this.name = name;
+        this.shopCode = shopCode;
+        this.createdAt = createdAt;
+        this.updatedAt = updatedAt;
+    }
+
+    /**
+     * Gets id.
+     *
+     * @return the id
+     */
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * Sets id.
+     *
+     * @param id the id
+     */
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    /**
+     * Gets code.
+     *
+     * @return the code
+     */
+    public String getCode() {
+        return code;
+    }
+
+    /**
+     * Sets code.
+     *
+     * @param code the code
+     */
+    public void setCode(String code) {
+        this.code = code;
+    }
+
+    /**
+     * Gets name.
+     *
+     * @return the name
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Sets name.
+     *
+     * @param name the name
+     */
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Gets shop code.
+     *
+     * @return the shop code
+     */
+    public String getShopCode() {
+        return shopCode;
+    }
+
+    /**
+     * Sets shop code.
+     *
+     * @param shopCode the shop code
+     */
+    public void setShopCode(String shopCode) {
+        this.shopCode = shopCode;
+    }
+
+    /**
+     * Gets created at.
+     *
+     * @return the created at
+     */
+    public Date getCreatedAt() {
+        return createdAt;
+    }
+
+    /**
+     * Sets created at.
+     *
+     * @param createdAt the created at
+     */
+    public void setCreatedAt(Date createdAt) {
+        this.createdAt = createdAt;
+    }
+
+    /**
+     * Gets upated at.
+     *
+     * @return the upated at
+     */
+    public Date getUpdatedAt() {
+        return updatedAt;
+    }
+
+    /**
+     * Sets upated at.
+     *
+     * @param updatedAt the upated at
+     */
+    public void setUpdatedAt(Date updatedAt) {
+        this.updatedAt = updatedAt;
+    }
+
+    /**
+     * Gets packaging.
+     *
+     * @return the packaging
+     */
+    public String getPackaging() {
+        return packaging;
+    }
+
+    /**
+     * Sets packaging.
+     *
+     * @param packaging the packaging
+     */
+    public void setPackaging(String packaging) {
+        this.packaging = packaging;
+    }
+
+    /**
+     * Gets product type.
+     *
+     * @return the product type
+     */
+    public List getProductType() {
+        return productType;
+    }
+
+    /**
+     * Sets product type.
+     *
+     * @param productType the product type
+     */
+    public void setProductType(List productType) {
+        this.productType = productType;
+    }
+
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
What is the purpose of `equals` and `hashcode` ovverridding here.. If needed later we can add it when we really need

Reply:
Do you want me to remove now?  It's just for future purposes.

Reply:
I recommend to remove it, let's add when it is really required.

Reply:
Done